packages/core/src/web/bridges/mcp-app-adapter.ts

```diff
@@ -5,9 +5,49 @@ import type {
   McpUiRequestDisplayModeRequest,
   McpUiRequestDisplayModeResult,
 } from "@modelcontextprotocol/ext-apps";
+import type {
+  CallToolRequest,
+  CallToolResult,
+} from "@modelcontextprotocol/sdk/types.js";
+
 import type { BridgeExternalStore } from "./hooks/types.js";
 import { McpAppBridge } from "./mcp-app-bridge.js";
-import type { BridgeInterface, Methods } from "./types.js";
+import type { BridgeInterface, CallToolResponse, Methods } from "./types.js";
+
+export const callTool = async <
+  ToolArgs extends Record<string, unknown> | null = null,
+  ToolResponse extends CallToolResponse = CallToolResponse,
+>(
+  name: string,
+  args: ToolArgs,
+): Promise<ToolResponse> => {
+  const bridge = McpAppBridge.getInstance();
+  const response = await bridge.request<CallToolRequest, CallToolResult>({
+    method: "tools/call",
+    params: {
+      name,
+      arguments: args ?? undefined,
+    },
+  });
+
+  const text: string[] = [];
+
+  response.content.forEach((content) => {
+    if (content.type === "text") {
+      text.push(content.text);
+    }
+  });
+
+  const result = text.join("\n");
```

Comment:
```suggestion
  const result = response.content.filter(({ type }) => type === "text").map(({ text }) => text).join("\n");
```

Reply:
Yeah I know but ts is the worst about that. Using a typeguard will be cleaner